lib/sift/type_validator.rb

```diff
@@ -1,7 +1,7 @@
 module Sift
   # TypeValidator validates that the incoming param is of the specified type
   class TypeValidator
-    RANGE_PATTERN = { format: { with: /\A.+(?:[^.]\.\.\.[^.]).+\z/, message: "must be a range" } }.freeze
+    RANGE_PATTERN = { format: { with: /\A.+(?:[^.]\.\.\.[^.]).+\z/, message: "must be a range" }, valid_date_range: true }.freeze
```

Comment:
I'm not 100% sure ranges are only used for dates. I think ranges could be in use for other data types. 

If so I think this could be a breaking change for any API that uses this. If so this would require a major version bump, and will make testing pretty difficult.

Reply:
@amayer171 currently RANGE_PATTERN is used only for dates. Renamed it to DATETIME_PATTERN to reflect its current meaning better